src/App.tsx

```diff
@@ -16,34 +19,76 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.none);
+  const [reverse, setReverse] = useState(false);
+
+  const resetButton =
+    sortBy === SortType.none && reverse === false ? false : true;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => {
+            if (sortBy !== SortType.alphabetically) {
+              return setSortBy(SortType.alphabetically);
+            }
+          }}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortBy !== SortType.alphabetically,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            if (sortBy !== SortType.length) {
+              return setSortBy(SortType.length);
+            }
+          }}
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortBy !== SortType.length,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() =>
+            reverse === false ? setReverse(true) : setReverse(false)
+          }
```

Comment:
Good, but a better form is:

```suggestion
          onClick={() => setReverse(value => !value)}
```

This is not critical though